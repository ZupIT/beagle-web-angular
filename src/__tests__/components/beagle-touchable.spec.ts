/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { TestBed, async, ComponentFixture, fakeAsync } from '@angular/core/testing'
import { BeagleTouchableComponent }
  from '../../components/beagle-touchable/beagle-touchable.component'
import { setAndCallHandler } from './mocks/test-mocks.spec'

let component: BeagleTouchableComponent
let fixture: ComponentFixture<BeagleTouchableComponent>

describe('BeagleTouchableComponent', () => {


  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        BeagleTouchableComponent,
      ],
    }).compileComponents()
    fixture = TestBed.createComponent(BeagleTouchableComponent)
    component = fixture.componentInstance
    component.onPress = jest.fn()
    fixture.detectChanges()
  }))

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot()
  })

  it('should call handlers and set Inputs', fakeAsync(() => {
    spyOn(component, 'handleClick').and.callThrough()

    setAndCallHandler('div', 'Testing', 'click', fixture)

    expect(component.onPress).toHaveBeenCalled()
  }))

})
