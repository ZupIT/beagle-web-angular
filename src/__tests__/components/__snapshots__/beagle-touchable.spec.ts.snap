// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeagleTouchableComponent should match snapshot 1`] = `
<beagle-touchable
  onPress={[Function Function]}
>
  <div
    class="touchable"
  />
</beagle-touchable>
`;
